lib/explorer/query.ex

```diff
@@ -299,18 +300,20 @@ defmodule Explorer.Query do
   and friends to convert queries into anonymous functions.
   See the moduledoc for more information.
   """
-  defmacro query(expression) do
-    df = df_var()
-
+  defmacro query(df, expression) do
     quote do
-      fn unquote(df) ->
-        unquote(traverse(expression, df))
+      # To prevent leaking imports, we wrap this operation in a function that we
+      # immediately execute.
+      fun = fn ->
+        unquote(traverse_root(expression, df))
       end
+
+      fun.()
```

Comment:
I feel like there's a smarter way to do this. 🤔

Reply:
[8edd4de](https://github.com/elixir-explorer/explorer/pull/989/commits/8edd4de8c4491a221fcee426705d256786956b7b)